types/mydecimal.go

```diff
@@ -1301,6 +1302,12 @@ func (d *MyDecimal) ToHashKey() ([]byte, error) {
 		// thus ErrTruncated may be raised, we can ignore it here.
 		err = nil
 	}
+	precBuff := make([]byte, 4)
+	digitBuff := make([]byte, 4)
+	binary.LittleEndian.PutUint32(precBuff, uint32(prec))
+	binary.LittleEndian.PutUint32(digitBuff, uint32(digitsFrac))
+	buf = append(buf, precBuff...)
+	buf = append(buf, digitBuff...)
```

Comment:
> ToBin converts decimal to its binary fixed-length representation
two representations of the same length can be compared with memcmp
with the correct -1/0/+1 result

It should be modified in `ToBin` method.
And some unit tests for comparison are better to be added.